spark-indexer/src/test/java/org/uniprot/store/spark/indexer/proteome/converter/DatasetProteomeEntryConverterTest.java

```diff
@@ -0,0 +1,335 @@
+package org.uniprot.store.spark.indexer.proteome.converter;
+
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.catalyst.expressions.GenericRowWithSchema;
+import org.apache.spark.sql.types.StructType;
+import org.junit.jupiter.api.Test;
+import org.uniprot.core.citation.CitationDatabase;
+import org.uniprot.core.citation.impl.JournalArticleBuilder;
+import org.uniprot.core.citation.impl.SubmissionBuilder;
+import org.uniprot.core.impl.CrossReferenceBuilder;
+import org.uniprot.core.proteome.GenomeAssemblyLevel;
+import org.uniprot.core.proteome.GenomeAssemblySource;
+import org.uniprot.core.proteome.ProteomeEntry;
+import org.uniprot.core.proteome.ProteomeType;
+import org.uniprot.core.proteome.impl.ComponentBuilder;
+import org.uniprot.core.proteome.impl.GenomeAnnotationBuilder;
+import org.uniprot.core.proteome.impl.GenomeAssemblyBuilder;
+import org.uniprot.core.proteome.impl.ProteomeEntryBuilder;
+import org.uniprot.core.uniprotkb.taxonomy.impl.TaxonomyBuilder;
+import org.uniprot.store.spark.indexer.common.util.RowUtils;
+import scala.Tuple2;
+import scala.collection.JavaConverters;
+import scala.collection.Seq;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.samePropertyValuesAs;
+import static org.uniprot.core.citation.SubmissionDatabase.EMBL_GENBANK_DDBJ;
+import static org.uniprot.store.spark.indexer.proteome.ProteomeRDDReader.*;
+
```

Comment:
Please increase the code coverage otherwise sonar quality gate will fail. The current code coverage of DatasetProteomeEntryConverter is 70%